dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/CalculBG.mo

```diff
@@ -0,0 +1,43 @@
+within Dynawo.Electrical.Controls.Voltage.StaticVarCompensator;
+
+/*
+* Copyright (c) 2015-2019, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+*/
+
+model CalculBG "Final calculation of the susceptance and the conductance"
+  import Modelica;
+  import Dynawo.Electrical.Controls.Voltage.StaticVarCompensator.Parameters;
+
+  extends Parameters.Params_CalculBG;
+
+  Modelica.Blocks.Interfaces.RealInput BVarPu "Variable susceptance of the static var compensator in p.u (base SNom)" annotation(
+    Placement(visible = true, transformation(origin = {-120, 60}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-110, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput GCstPu "Fixed conductance of the static var compensator in p.u (base SNom)" annotation(
+    Placement(visible = true, transformation(origin = {-120, -60}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-110, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+  Modelica.Blocks.Interfaces.RealOutput BPu "Final susceptance of the static var compensator in p.u (base SNom)" annotation(
+    Placement(visible = true, transformation(extent = {{100, 50}, {120, 70}}, rotation = 0), iconTransformation(origin = {107, 59}, extent = {{-7, -7}, {7, 7}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealOutput GPu "Final conductance of the static var compensator in p.u (base SNom)" annotation(
+    Placement(visible = true, transformation(extent = {{100, -70}, {120, -50}}, rotation = 0), iconTransformation(origin = {107, -59}, extent = {{-7, -7}, {7, 7}}, rotation = 0)));
+
+  ModeConnector mode "Current mode of the static var compensator" annotation(
+    Placement(visible = true, transformation(origin = {62, 120}, extent = {{-20, -20}, {20, 20}}, rotation = -90), iconTransformation(origin = {72, 110}, extent = {{-10, -10}, {10, 10}}, rotation = -90)));
+
+equation
+
+  BPu = if mode.value == Mode.RUNNING_V then BVarPu + BShuntPu elseif mode.value == Mode.STANDBY then BShuntPu else 0;
+
+  GPu = if mode.value == Mode.OFF then 0 else GCstPu;
+
```

Comment:
Open question again: basically you have two ways of writing these two if-loops. Either separately as you have done or in a unique loop such as:
if mode.value == Mode.RUNNING_V then
 BPu = BVarPu + BShuntPu;
 GPu = 0;
elseif 
 ....

In the second case, you see the overall SVarC behavior in each mode at first glance but if you prefer the current implementation, you can stick to it.

Reply:
good idea, fixed